libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/CloudAdapter.cs

```diff
@@ -53,54 +58,219 @@ public CloudAdapter()
+                _adapter = adapter;
+
+                // Internal reuse of the existing StreamingRequestHandler class
+                _requestHandler = new StreamingRequestHandler(bot, this, pipeName, _authenticateRequestResult.Audience, adapter.Logger);
+
+                // Fix up the connector factory so connector create from it will send over this connection
+                _authenticateRequestResult.ConnectorFactory = new StreamingConnectorFactory(_requestHandler);
             }
-            else
+
+            public Task ListenAsync() => _requestHandler.ListenAsync();
+
+            Task<InvokeResponse> IStreamingActivityProcessor.ProcessStreamingActivityAsync(Activity activity, BotCallbackHandler callback, CancellationToken cancellationToken)
+                => _adapter.ProcessActivityAsync(_authenticateRequestResult, activity, callback, cancellationToken);
+
+            private class StreamingConnectorFactory : ConnectorFactory
             {
-                // Deserialize the incoming Activity
-                var activity = await HttpHelper.ReadRequestAsync<Activity>(httpRequest).ConfigureAwait(false);
+                private readonly StreamingRequestHandler _requestHandler;
+                private string _serviceUrl = null;
 
-                if (string.IsNullOrEmpty(activity?.Type))
+                public StreamingConnectorFactory(StreamingRequestHandler requestHandler)
                 {
-                    httpResponse.StatusCode = (int)HttpStatusCode.BadRequest;
-                    return;
+                    _requestHandler = requestHandler;
                 }
 
-                // Grab the auth header from the inbound http request
-                var authHeader = httpRequest.Headers["Authorization"];
-
-                try
+                public override Task<IConnectorClient> CreateAsync(string serviceUrl, string audience, CancellationToken cancellationToken)
                 {
-                    // Process the inbound activity with the bot
-                    var invokeResponse = await ProcessActivityAsync(authHeader, activity, bot.OnTurnAsync, cancellationToken).ConfigureAwait(false);
+                    if (_serviceUrl == null)
+                    {
+                        _serviceUrl = serviceUrl;
+                    }
+                    else if (!_serviceUrl.Equals(serviceUrl, StringComparison.OrdinalIgnoreCase))
+                    {
+                        throw new NotSupportedException("This is a streaming scenario, all connectors from this factory must all be for the same url.");
+                    }
 
-                    // write the response, potentially serializing the InvokeResponse
-                    await HttpHelper.WriteResponseAsync(httpResponse, invokeResponse).ConfigureAwait(false);
+#pragma warning disable CA2000 // Dispose objects before losing scope
+                    var streamingHttpClient = new StreamingHttpClient(_requestHandler);
```

Comment:
This is never disposed?

Reply:
here it's just the abstraction no underlying resource to clean up - if you want something to be concerned about take a look at the disposable members of the StteamingRequestHandler taht are not disposed